addons/website_sale/static/src/js/website_sale_utils.js

```diff
@@ -32,7 +36,47 @@ function animateClone($cart, $elem, offsetTop, offsetLeft) {
     }
 }
 
+/**
+ * Allows to propose the existing products from the database.
+ *
+ * @param {ServicesMixin|Widget} self - an element capable to trigger an RPC
+ * @param {jQuery} $input
+ */
+function autocompleteWithProducts(self, $input, data) {
+    $input.autocomplete({
+        source: function (request, response) {
+            return self._rpc({
+                route: '/website_sale/get_products_info',
+                params: {
+                    term: request.term,
+                    limit: 7,
+                    product_options: {
+                        'display_description': data.displayDescription,
+                        'display_price': data.displayPrice,
+                        'display_image': data.displayImage,
+                        'order_by': data.orderBy,
+                    }
+                },
+            }).then((function (resources) {
+                this.products_options = resources.products_options;
+                this.products_info = resources.products_info;
+                response(this.products_info);
+            }).bind(this));
+        },
+    }).autocomplete('instance')._renderItem = function (ul, item) {
+        var priceWithCurrency = item.currency_position === 'after' ? item.price + ' ' + item.currency_symbol : item.currency_symbol + ' ' + item.price;
+        var params = {
+            'products_options': this.products_options,
+            'item': item,
+            'price_with_currency': priceWithCurrency,
+        };
+        var $li = $(qweb.render('website_sale.productsSearchBar.autocompleteItem', params));
```

Comment:
`var $li = $(...)` and it is a div I see in the qweb template... I think it is supposed to be a `<li>`?